pkg/util/extender/unstructured.go

```diff
@@ -261,21 +268,48 @@ func SetLastAppliedConfigAnnotation(u UnstructuredExtender, obj *unstructured.Un
 	return nil
 }
 
-func SetUpgradePartition(u UnstructuredExtender, obj *unstructured.Unstructured, upgradeOrdinal, gracePeriod int64, advanced bool) error {
-	if err := u.SetSpecField(obj, "RollingUpdate", "updateStrategy", "type"); err != nil {
+func SetUpdatePartition(
+	extender UnstructuredExtender,
+	obj *unstructured.Unstructured,
+	upgradeOrdinal,
+	gracePeriod int64,
+	advanced bool) error {
+	if err := extender.SetSpecField(obj, "RollingUpdate", "updateStrategy", "type"); err != nil {
 		return err
 	}
-	if err := u.SetSpecField(obj, upgradeOrdinal, "updateStrategy", "rollingUpdate", "partition"); err != nil {
+	if err := extender.SetSpecField(obj, upgradeOrdinal, "updateStrategy", "rollingUpdate", "partition"); err != nil {
 		return err
 	}
 	if advanced {
-		if err := u.SetSpecField(obj, "InPlaceIfPossible", "updateStrategy", "rollingUpdate", "podUpdatePolicy"); err != nil {
+		if err := extender.SetSpecField(obj, "InPlaceIfPossible", "updateStrategy", "rollingUpdate", "podUpdatePolicy"); err != nil {
 			return err
 		}
-		if err := u.SetSpecField(obj, gracePeriod,
+		if err := extender.SetSpecField(obj, gracePeriod,
 			"updateStrategy", "rollingUpdate", "inPlaceUpdateStrategy", "gracePeriodSeconds"); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func SetContainerImage(obj *unstructured.Unstructured, containerName, image string) error {
```

Comment:
The others uses the extender parameter. Why isn't this one? Are there any considerations?

Reply:
Refactored